src/Course/Monad.hs

```diff
@@ -32,48 +29,39 @@ infixr 1 =<<
 -- >>> (\x -> ExactlyOne(x+1)) =<< ExactlyOne 2
 -- ExactlyOne 3
 instance Monad ExactlyOne where
-  (=<<) ::
-    (a -> ExactlyOne b)
-    -> ExactlyOne a
-    -> ExactlyOne b
-  (=<<) =
-    error "todo: Course.Monad (=<<)#instance ExactlyOne"
+  (=<<) :: (a -> ExactlyOne b) -> ExactlyOne a -> ExactlyOne b
+  f =<< (ExactlyOne x) = f x
+
+-- Left identity: return a >>= f ≡ f a
 
+-- Right identity: m >>= return ≡ m
+
+-- Associativity: (m >>= f) >>= g ≡ m >>= (\x -> f x >>= g)
 -- | Binds a function on a List.
 --
 -- >>> (\n -> n :. n :. Nil) =<< (1 :. 2 :. 3 :. Nil)
 -- [1,1,2,2,3,3]
 instance Monad List where
-  (=<<) ::
-    (a -> List b)
-    -> List a
-    -> List b
-  (=<<) =
-    error "todo: Course.Monad (=<<)#instance List"
+  (=<<) :: (a -> List b) -> List a -> List b
+  f =<< xs = flatten . map f $ xs
```

Comment:
Nice! we also have `concatMap`, so `f =<< xs = concatMap f xs` which... we can rewrite as
`(=<<) = concatMap` ;-)

Reply:
we have that one? that was going to be my first approach but I thought it wasn't defined

Reply:
Ah it might be called `flatMap`... but yeah I think so... it's in `List`, IIRC. But either way, it's good 👍